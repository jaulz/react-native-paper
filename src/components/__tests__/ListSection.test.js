/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import ListSection from '../List/ListSection';
import ListItem from '../List/ListItem';
import ListIcon from '../List/ListIcon';

it('renders list section without title', () => {
  const tree = renderer
    .create(
      <ListSection>
        <ListItem
          title="First Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
        <ListItem
          title="Second Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list section with title', () => {
  const tree = renderer
    .create(
      <ListSection title="Some title">
        <ListItem
          title="First Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
        <ListItem
          title="Second Item"
          left={props => <ListIcon {...props} icon="folder" />}
        />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
