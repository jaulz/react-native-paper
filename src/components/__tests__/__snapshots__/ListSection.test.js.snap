// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list section with title 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "marginHorizontal": 16,
            "marginVertical": 13,
          },
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "HelveticaNeue-Medium",
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "backdrop": "rgba(0, 0, 0, 0.5)",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.54)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    Some title
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#03dac4",
                "backdrop": "rgba(0, 0, 0, 0.5)",
                "background": "#f6f6f6",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "error": "#B00020",
                "placeholder": "rgba(0, 0, 0, 0.54)",
                "primary": "#6200ee",
                "surface": "#ffffff",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 4,
            }
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#03dac4",
                "backdrop": "rgba(0, 0, 0, 0.5)",
                "background": "#f6f6f6",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "error": "#B00020",
                "placeholder": "rgba(0, 0, 0, 0.54)",
                "primary": "#6200ee",
                "surface": "#ffffff",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 4,
            }
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list section without title 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#03dac4",
                "backdrop": "rgba(0, 0, 0, 0.5)",
                "background": "#f6f6f6",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "error": "#B00020",
                "placeholder": "rgba(0, 0, 0, 0.54)",
                "primary": "#6200ee",
                "surface": "#ffffff",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 4,
            }
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#03dac4",
                "backdrop": "rgba(0, 0, 0, 0.5)",
                "background": "#f6f6f6",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "error": "#B00020",
                "placeholder": "rgba(0, 0, 0, 0.54)",
                "primary": "#6200ee",
                "surface": "#ffffff",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 4,
            }
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;
